lexer.go

```diff
@@ -376,8 +378,13 @@ func startWithSlash(s *Scanner) (tok int, pos Pos, lit string) {
 		// See https://dev.mysql.com/doc/refman/5.7/en/optimizer-hints.html
 		if _, ok := hintedTokens[s.lastKeyword]; ok {
 			// only recognize optimizers hints directly followed by certain
-			// keywords like SELECT, INSERT, etc.
-			isOptimizerHint = true
+			// keywords like SELECT, INSERT, etc., only a special case "FOR UPDATE" needs to be handled
+			// we will report a warning in order to match MySQL's behavior, but the hint content will be ignored
```

Comment:
note the MySQL doesn't necessarily warn, e.g. the following produces no warnings (and the hint is ignored too).

```sql
select 1 for update /*+ hash_join() */;
```

but we don't need to replicate this MySQL behavior. the current solution is good enough.